speaker/speaker_recognition_result.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+package speaker
+
+import (
+	"strconv"
+	"unsafe"
+
+	"github.com/Microsoft/cognitive-services-speech-sdk-go/common"
+)
+
+// #include <stdlib.h>
+// #include <stdint.h>
+// #include <speechapi_c_common.h>
+// #include <speechapi_c_result.h>
+// #include <speechapi_c_recognizer.h>
+//
+import "C"
+
+// SpeakerRecognitionResult contains information about result from voice profile operations.
+type SpeakerRecognitionResult struct {
+	handle C.SPXHANDLE
+
+	// ResultID specifies the result identifier.
+	ResultID string
+
+	// Reason specifies status of speech synthesis result.
+	Reason common.ResultReason
+
+	// ProfileID specifies the recognized profile id.
+	ProfileID string
+
+	// Score specifies the confidence score for the recognized profile id.
+	Score float64
+
+	// ErrorDetails presents error details.
+	ErrorDetails string
+
+	// Collection of additional properties.
+	Properties *common.PropertyCollection
+}
+
+// Close releases the underlying resources
+func (result SpeakerRecognitionResult) Close() {
+	result.Properties.Close()
+	C.recognizer_result_handle_release(result.handle)
+}
+
+// NewSpeakerRecognitionResultFromHandle creates a SpeakerRecognitionResult from a handle (for internal use)
+func NewSpeakerRecognitionResultFromHandle (handle common.SPXHandle) (*SpeakerRecognitionResult, error) {
```

Comment:
this needs to be public?

Reply:
used in speaker_recognizer

Reply:
Didn't realize private meant package level, not class level. Will fix all.